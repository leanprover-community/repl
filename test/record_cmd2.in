{"cmd": "example  : 2=2 := by"}

{"cmd": "example  : 2=2 := by\nrfl"}

{"cmd": "theorem gc_demo (P Q R : Prop) : (P → Q) → (Q → R) → (P → R) := by", "record": false}

{"cmd": "theorem gc_demo (P Q R : Prop) : (P → Q) → (Q → R) → (P → R) := by\nintro h1 h2 p", "record": false}

{"cmd": "theorem gc_demo (P Q R : Prop) : (P → Q) → (Q → R) → (P → R) := by\nintro h1 h2 p\napply?", "record": false}
