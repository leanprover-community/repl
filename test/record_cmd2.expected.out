{"messages":
 [{"severity": "error",
   "pos": {"line": 1, "column": 20},
   "endPos": null,
   "data": "unexpected end of input; expected '{'"},
  {"severity": "error",
   "pos": {"line": 1, "column": 18},
   "endPos": {"line": 1, "column": 20},
   "data": "unsolved goals\n⊢ 2 = 2"}],
 "env": 0}

{"env": 1}

{"messages":
 [{"severity": "error",
   "pos": {"line": 1, "column": 66},
   "endPos": null,
   "data": "unexpected end of input; expected '{'"},
  {"severity": "error",
   "pos": {"line": 1, "column": 64},
   "endPos": {"line": 1, "column": 66},
   "data": "unsolved goals\nP Q R : Prop\n⊢ (P → Q) → (Q → R) → P → R"}]}

{"messages":
 [{"severity": "error",
   "pos": {"line": 1, "column": 64},
   "endPos": {"line": 2, "column": 13},
   "data": "unsolved goals\nP Q R : Prop\nh1 : P → Q\nh2 : Q → R\np : P\n⊢ R"}]}

{"messages":
 [{"severity": "info",
   "pos": {"line": 3, "column": 0},
   "endPos": {"line": 3, "column": 6},
   "data": "Try this: exact h2 (h1 p)"}]}

